install.sh

```diff
@@ -0,0 +1,253 @@
+#
+# Copyright (C) 2021 CloudTruth, Inc.
+#
+
+set -e
+
+### Control     ############################################################
+
+CT_CLI_VERSION=
+CT_DRAFT_AUTH_TOKEN=
+CT_DRAFT_RELEASE_ID=
+CT_DRY_RUN=0
+CT_INSTALL_PREREQUISITES=1
+
+### Detection     ############################################################
+
+# detect the architecture
+ARCH=$(uname -m)
+
+if [ -z "${ARCH}" ]; then
+    >&2 echo "Cannot determine system architecture."
+    exit 42
+fi
+
+# detect which package type the system wants
+OS=$(uname)
+PKG=
+if [ "${OS}" = "Linux" ]; then
+    command -v apk && PKG=apk
+    command -v apt-get && PKG=deb
+    command -v yum && PKG=rpm
+
+elif [ "${OS}" = "Darwin" ]; then
+    PKG=macos
+fi
+
+if [ -z "${PKG}" ]; then
+    >&2 echo "[error] cannot determine system package format"
+    exit 43
+fi
+
+### Arguments     ############################################################
+
+while true; do
+    case $1 in
+      (-a|--auth-token)
+            CT_DRAFT_AUTH_TOKEN=$2
+            shift 2;;
+      (-d|--debug)
+            echo "[debug] enabled"
+            EXTRA_CURL_OPTIONS="-v"
+            set -x
+            shift;;
+      (-y|--dry-run)
+            CT_DRY_RUN=1
+            echo "[dry-run] enabled"
+            shift;;
+      (-h|--help)
+            echo "Usage: install.sh [ OPTIONS ]"
+            echo ""
+            echo "OPTIONS:"
+            echo ""
+            echo "  -d | --debug             enable shell debug output"
+            echo "  -h | --help              show usage"
+            echo "  -n | --no-prerequisites  do not attempt to install prerequisites"
+            echo "  -v | --version <VER>     use a specific version"
+            echo "  -y | --dry-run           download but do not install (may fail if prerequisites are missing)"
+            echo ""
+            echo "These options are only used for testing during the CloudTruth release workflow:"
+            echo ""
+            echo "  -a | --auth-token <TOK>  authorization token to access draft release"
+            echo "  -r | --release-id <ID>   identity of draft release"
+            echo ""
+            exit 2;;
+      (-n|--no-prerequisites)
+            CT_INSTALL_PREREQUISITES=0
+            shift;;
+      (-r|--release-id)
+            CT_DRAFT_RELEASE_ID=$2
+            shift 2;;
+      (-v|--version)
+            CT_CLI_VERSION=$2
+            shift 2;;
+      (--)  shift; break;;
+      (*)
+            if [ -n "${1}" ]; then
+                echo "[error] invalid parameter: ${1}"
+                exit 1; # error
+            fi
+            break;;
+    esac
+done
+# remaining="$@"
+
+### Prerequisites ############################################################
+
+PREREQUISITES="curl"
+if [ -n "${CT_DRAFT_RELEASE_ID}" ]; then
+  # additional requirements to handle GitHub draft release integration testing
+  PREREQUISITES="${PREREQUISITES} ca-certificates jq"
+fi
+
+# shellcheck disable=SC2086
+prerequisites() {
```

Comment:
I'm not aware of any.  Tests pass.

Reply:
I will try in a while, but I don't think I will be able to run in my environment (since I don't have `jq` installed). I'm not sure how/why it works in the pipeline.

Reply:
The macos runner in GHA has jq and a bunch of other things installed... I didn't make a docker container to isolate it (not sure you can yet...)